sale_exception/tests/test_sale_exception.py

```diff
@@ -204,3 +204,16 @@ def test_exception_no_free(self):
         so_except_confirm.action_confirm()
         self.assertFalse(sale_order.ignore_exception)
         self.assertTrue(sale_order.state == "draft")
+
+    def test_exception_no_stock_warning(self):
+        exception = self.env.ref("sale_exception.excep_no_stock")
+        exception.active = True
+        partner = self.env.ref("base.res_partner_1")
+        sale_order = self._create_sale_order(
+            partner=partner, product=self.env.ref("product.product_product_6")
+        )
+        sale_order.action_confirm()
+        sale_order.action_done()
+        self.assertEqual(sale_order.state, "done")
+        sale_order.action_unlock()
+        self.assertEqual(sale_order.state, "sale")
```

Comment:
This test scenario doesn't look valid.  We may first process the SO up to done, then activate an exception rule that should raise an exception without the `check_exception=False` context.

Reply:
Hi!

Thanks for your suggestion! I have change the scenario as you proposed.

Firstly, the SO is created and confirmed, after that, the exception rule is activated (It is suppose to raise because the partner does not have any zip code). Secondly, the SO is locked and unlock, without any exception.

Regards